src/platform/android/DnssdImpl.cpp

```diff
@@ -213,7 +234,26 @@ void InitializeWithObjects(jobject resolverObject, jobject mdnsCallbackObject)
     }
 }
 
-void HandleResolve(jstring instanceName, jstring serviceType, jstring address, jint port, jlong callbackHandle, jlong contextHandle)
+/**
+ * Helper function which returns the jstring key as a null-terminated char string for the purpose
```

Comment:
```cpp
/// Returns a static `const char *` string that is equivalent to the passed in jstring key
/// so that the jstring can be released while the key value remains valid.
///
/// Only supports known Matter DNSSD TXT key values, returns empty string otherwise.
```